pkg/avatar/avatar.go

```diff
@@ -30,10 +33,199 @@ func (p *Player) Type() targets.TargettableType { return targets.TargettablePlay
 
 func (p *Player) HandleAttack(atk *combat.AttackEvent) float64 {
 	p.Core.Combat.Events.Emit(event.OnPlayerHit, p, atk)
+	var amp string
+	var cata string
+	var dmg float64
+	var crit bool
+	evt := p.Core.Combat.Log.NewEvent(atk.Info.Abil, glog.LogDamageEvent, atk.Info.ActorIndex)
+
+	// TODO: Implement reactions on player
+
+	dmg, crit = p.calc(atk, evt)
 
-	//TODO: implement player taking damage here
+	active := p.Core.Player.Active()
+	dmgLeft := p.Core.Player.Shields.OnDamage(active, dmg, atk.Info.Element)
+	if dmgLeft > 0 {
+		p.Core.Player.Drain(player.DrainInfo{
+			ActorIndex: active,
+			Abil:       atk.Info.Abil,
+			Amount:     dmgLeft,
+		})
+	}
+	evt.Write("target", p.Key()).
+		Write("attack-tag", atk.Info.AttackTag).
+		Write("ele", atk.Info.Element.String()).
+		Write("damage", &dmgLeft).
+		Write("damage_before_shield", &dmg).
```

Comment:
Please swap these two lines (makes more sense to me to see dmg before shield first).